CHANGELOG.md

```diff
@@ -4,6 +4,24 @@
 <!--- Distributed under the Boost Software License, Version 1.0. (See accompanying -->
 <!--- file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt) -->
 
+## 0.19.0 (2023-10-02)
+
+### New features
+
+- A `transfer_when_all` sender adaptor has been introduced. ([#792](https://github.com/pika-org/pika/pull/792))
+- We are now testing pika compiled with C++23. ([#779](https://github.com/pika-org/pika/pull/779))
```

Comment:
CI changes in themselves are not a new feature. In my opinion this can be left out but if you think it's worth mentioning we should maybe introduce a new section for miscellaneous changes like this?

Reply:
Since this doesn't have any functional impact for users I consider this non-news. If we add conditional support for some C++23 feature that would be worth adding here in my opinion.